content-src/asrouter/templates/TrailheadOverlay/TrailheadOverlay.jsx

```diff
@@ -0,0 +1,231 @@
+    document.body.classList.remove("hide-main", "trailhead-fxa");
+    this.setState({ show: false });
+    // Re-enable the document for screen readers
+    this.props.document
+      .getElementById("root")
+      .setAttribute("aria-hidden", "false");
+
+    setTimeout(() => {
+      // Allow scrolling and fully remove overlay after animation finishes.
+      this.props.onBlock();
+      document.body.classList.remove("welcome");
+    }, 400);
+  }
+
+  onInputChange(e) {
+    let error = e.target.previousSibling;
+    this.setState({ emailInput: e.target.value });
+    error.classList.remove("active");
+    e.target.classList.remove("invalid");
+  }
+
+  onSubmit() {
+    this.props.dispatch(
+      ac.UserEvent({ event: "SUBMIT_EMAIL", ...this._getFormInfo() })
+    );
+
+    window.addEventListener("visibilitychange", this.removeOverlay);
+  }
+
+  clickSkip() {
+    this.props.dispatch(
+      ac.UserEvent({ event: "SKIPPED_SIGNIN", ...this._getFormInfo() })
+    );
+    this.removeOverlay();
+  }
+
+  /**
+   * Report to telemetry additional information about the form submission.
+   */
+  _getFormInfo() {
+    const value = {
+      has_flow_params: this.props.flowParams.flowId.length > 0,
+    };
+    return { value };
+  }
+
+  onInputInvalid(e) {
+    let error = e.target.previousSibling;
+    error.classList.add("active");
+    e.target.classList.add("invalid");
+    e.preventDefault(); // Override built-in form validation popup
+    e.target.focus();
+  }
+
+  render() {
+    // eslint was getting a false positive caused by the dynamic injection
+    // of content.
+    // eslint-disable-next-line jsx-a11y/anchor-has-content
+    return (
+      <div className={`overlay-wrapper ${this.state.show ? "show" : ""}`}>
+        <div className="background" />
+        <div className="firstrun-scene">
+          <div className="fxaccounts-container">
+            <div className="firstrun-left-divider">
+              <h1
+                className="firstrun-title"
+                data-l10n-id="onboarding-sync-welcome-header"
+              />
+              <p
+                className="firstrun-content"
+                data-l10n-id="onboarding-sync-welcome-content"
+              />
+              <a
+                className="firstrun-link"
+                href={`https://www.mozilla.org/firefox/features/sync/?${
+                  this.utmParams
+                }`}
+                target="_blank"
+                rel="noopener noreferrer"
+                data-l10n-id="onboarding-sync-welcome-learn-more-link"
+              />
+            </div>
+            <div className="firstrun-sign-in">
+              <p className="form-header">
+                <span data-l10n-id="onboarding-sync-form-header" />
+                <span
+                  className="sub-header"
+                  data-l10n-id="onboarding-sync-form-sub-header"
+                />
+              </p>
+              <form
```

Comment:
As we discussed, let's refactor the `Trailhead` component to have a `FXASignupForm` component inside of it that can be reused in this template.